src/mui/withChildrenAsFunction.js

```diff
@@ -15,23 +14,27 @@ export default BaseComponent => {
         };
 
         componentDidMount() {
-            this.initializeChildren(this.props.children);
+            return this.initializeChildren(this.props.children);
```

Comment:
why ?

Reply:
Because we want to return the promise of the async function `initializeChildren` so that it is handled by React lifecycle

Reply:
got it